Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -0,0 +1,1129 @@
+                                           filestream.Length,
+                                           contentType);
+
+                await minio.GetObjectAsync(bucketName, objectName,
+              (stream) =>
+              {
+                  var fileStream = File.Create(tempFileName);
+                  stream.CopyTo(fileStream);
+                  fileStream.Dispose();
+                  FileInfo writtenInfo = new FileInfo(tempFileName);
+                  file_read_size = writtenInfo.Length;
+
+                  Assert.AreEqual(file_read_size, file_write_size);
+                  File.Delete(tempFileName);
+              });
+
+                await minio.RemoveObjectAsync(bucketName, objectName);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("[Bucket]  Exception: {0}", e);
+                Assert.Fail();
+            }
+            await TearDown(minio, bucketName);
+            File.Delete(fileName);
+            Console.Out.WriteLine("Test1: GetObjectAsync Complete");
+        }
+        private async static Task GetObject_Test2(MinioClient minio)
+        {
+            Console.Out.WriteLine("Test2: GetObjectAsync for non existent object");
+            string bucketName = GetRandomName(15);
+            string objectName = GetRandomName(10);
+            string fileName = GetRandomName(10);
+            await Setup_Test(minio, bucketName);
+            try
+            {
+                await minio.GetObjectAsync(bucketName, objectName, fileName);
+
+            }
+            catch (ObjectNotFoundException ex)
+            {
+                Assert.AreEqual(ex.message, "Not found.");
+            }
+
+            await TearDown(minio, bucketName);
+            Console.Out.WriteLine("Test2: GetObjectAsync Complete");
+        }
+        private async static Task GetObject_Test3(MinioClient minio)
+        {
+            Console.Out.WriteLine("Test3: GetObjectAsync for partial object from stream");
+            string bucketName = GetRandomName(15);
+            string objectName = GetRandomName(10);
+            string fileName = CreateFile(1 * MB);
+            string contentType = null;
+            await Setup_Test(minio, bucketName);
+            try
+            {
+                byte[] bs = File.ReadAllBytes(fileName);
+                System.IO.MemoryStream filestream = new System.IO.MemoryStream(bs);
+                long file_write_size = 10L;
+                string tempFileName = "tempFileName";
+                long file_read_size = 0;
+                await minio.PutObjectAsync(bucketName,
+                                           objectName,
+                                           filestream,
+                                           filestream.Length,
+                                           contentType);
+
+                await minio.GetObjectAsync(bucketName, objectName, 1024L, file_write_size,
+              (stream) =>
+              {
```

Comment:
The following snippet of code is repeated in a lot of places. Can we move it to a common utils class?